arcaflow_plugin_rtla/rtla_plugin.py

```diff
@@ -130,53 +130,62 @@ def run_timerlat(
             # Capture the column headers
-            elif re_isindex.match(line):
+            elif is_header.match(line):
                 col_headers = line.lower().split()
-            # Stats names repeat, so flag when have passed ^ALL
-            elif re_isall.match(line):
-                found_all = True
-            # Either this is a histogram bucket row, or the first time we have seen
-            # a row beginning with a stat name
-            elif (re_isdigit.match(line)) or (
-                line.split()[0] in stats_names and not found_all
-            ):
+                line_num += 1
+                break
+
+        # Phase 2: Get the columnar data
+        for i in range(line_num, len(output_lines)):
+            line_list = output_lines[i].split()
+            row_obj = {}
+            # Collect histogram buckets and column latency statistics
+            if not is_summary.match(line_list[0]):
                 # Capture the columnar data
-                line_list = []
-                for element in line.split():
-                    try:
-                        line_list.append(int(element))
-                    except ValueError:
-                        line_list.append(element)
-                row_obj = dict(zip(col_headers, line_list))
-                if re_isdigit.match(line):
-                    latency_hist.append(row_obj)
+                if not is_digit.match(line_list[0]):
+                    # Stats index values are strings
+                    row_obj[col_headers[0]] = line_list[0][:-1]
+                    accumulator = stats_per_col
                 else:
-                    stats_per_col.append(row_obj)
-            # Since we've encountered the summary statistics (marked by the line
-            # starting with "ALL"), generate key:value pairs instead of columnar data.
-            elif found_all and line.split()[0] in stats_names:
-                label = line.split()[0][:-1]
-                if label != "over":
-                    total_irq_latency[label] = line.split()[1]
-                    total_thr_latency[label] = line.split()[2]
-                    if params.user_threads:
-                        total_usr_latency[label] = line.split()[3]
+                    # Histogram index values are integers
+                    row_obj[col_headers[0]] = int(line_list[0])
+                row_obj = row_obj | dict(zip(col_headers[1:], map(int, line_list[1:])))
+                accumulator.append(row_obj)
+            else:
+                line_num = i + 1
+                break
+
+        # Phase 3: Get the stats summary as key:value pairs
+        for i in range(line_num, len(output_lines)):
+            line_list = output_lines[i].split()
+            label = line_list[0][:-1]
+            total_irq_latency[label] = line_list[1]
+            total_thr_latency[label] = line_list[2]
+            if params.user_threads:
+                total_usr_latency[label] = line_list[3]
```

Comment:
Since you only want to visit each line once, I suggest using an iterator instead of enumeration:
```suggestion
        output_lines = iter(output.splitlines())

        # Phase 1: Get the headers
        for line in output_lines:
            # Get the time unit (user-selectable)
            if is_time_unit.match(line):
                time_unit = is_time_unit.match(line).group(1)
            # Capture the column headers
            elif is_header.match(line):
                col_headers = line.lower().split()
                break

        # Phase 2: Collect histogram buckets and column latency statistics
        for line in output_lines:
            line_list = line.split()
            # Collect statistics up until the summary section
            # (We don't process the summary header line itself, we just skip it here.)
            if is_summary.match(line_list[0]):
                break

            row_obj = dict(zip(col_headers[1:], map(int, line_list[1:])))
            if not is_digit.match(line_list[0]):
                # Stats index values are strings ending in a colon
                row_obj[col_headers[0]] = line_list[0][:-1]
                # When we hit the stats, switch to the other accumulator
                accumulator = stats_per_col
            else:
                # Histogram index values are integers
                row_obj[col_headers[0]] = int(line_list[0])
            accumulator.append(row_obj)

        # Phase 3: Get the stats summary as key:value pairs
        for line in output_lines:
            line_list = line.split()
            label = line_list[0][:-1]
            total_irq_latency[label] = line_list[1]
            total_thr_latency[label] = line_list[2]
            if params.user_threads:
                total_usr_latency[label] = line_list[3]
```
That is, using the iterator allows us to avoid enumerating the output lines and having to keep track of where we are; we process the lines in sequence, moving to the next phase like a state machine.

Also, I tweaked Phase 2.  Moving the `break` up allows us to avoid the extra level of indentation/complexity.  And, using the result of the `zip()` call as the initializer for `row_obj` saves us from having to use the "update" operation to add it to `row_obj` later.  And, I added/reworked the comments, slightly.

_[Disclaimer:  I didn't actually try running this code....]_

Reply:
I'll play with this as an alternative to the `.pop()`, suggested above by Dave, that was giving me fits.